### PR TITLE
feat(components): document adding lint rules to desktop modeler

### PR DESCRIPTION
**Pull request by philippfromme**
Related to https://github.com/camunda/camunda-modeler/issues/3269

## What is the purpose of the change

Camunda Modeler 5.4 supports adding lint rules for the new built-in linting feature (introduced in Camunda Modeler 5.0). Lint rules can be added through a Camunda Modeler plugin which is [already documented](https://docs.camunda.io/docs/next/components/modeler/desktop-modeler/plugins/) but since it's a special kind of plugin we want to add a dedicated section for it. That section is also [linked in the general plugins section](https://docs.camunda.io/docs/next/components/modeler/desktop-modeler/plugins/#overview-of-your-possibilities-as-a-plugin-developer).

## Are there related marketing activities

No further activities.

## When should this change go live?

Immediately, as this feature has already been released.

## PR Checklist

- [ ] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [ ] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+[bpmnlint](https://github.com/bpmn-io/bpmnlint) plugins allow you to add custom lint rules for BPMN diagrams or disable existing rules.
+
+## Getting started
+
+To get started, you can use [this example project](https://github.com/camunda/camunda-modeler-custom-linter-rules-plugin).
+
+Take the following steps:
+
+1. Clone the repository:
```
Comment (barmac) [inline docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:15]: If we want to encourage people to share their work, I'd rather suggest to fork the repo instead.

File: docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
+---
+id: lint-rules
+title: Lint rules
+description: "Lint rule plugins allow you to add your own lint rules to the Desktop Modeler."
```
Comment (nikku) [inline docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:4]:
```suggestion
description: "Learn how to contribute additional validation (lint) rules to the Desktop Modeler through lint rule plug-ins."
```

File: docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+---
+
+[bpmnlint](https://github.com/bpmn-io/bpmnlint) plugins allow you to add custom lint rules for BPMN diagrams or disable existing rules.
```
Comment (nikku) [inline docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:7]:
```suggestion
The Desktop Modeler uses [bpmnlint](https://github.com/bpmn-io/bpmnlint) to validate BPMN diagrams. Through defined modeler extension points you add plug-in additional lint rules or configure/disable existing rules.
```

File: docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+## Getting started
+
+To get started, you can use [this example project](https://github.com/camunda/camunda-modeler-custom-linter-rules-plugin).
```
Comment (nikku) [inline docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:11]:
```suggestion
To get started, you can start from the [custom-linter-rules-plugin template)[https://github.com/camunda/camunda-modeler-custom-linter-rules-plugin).
```

**Review by nikku — COMMENTED** [review]
Some cosmetic suggestions from my end; looks nice! :)

**Comment by philippfromme** [comment]
This pull request has been reviewed internally in the Desktop Modeler and bpmn.io team. It's ready to be reviewed and merged. @akeller @christinaausley @pepopowitz 👀

File: docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
+---
+id: lint-rules
+title: Lint rules
```
Comment (christian-konrad) [inline docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:3]: What about "Custom lint rules"?

File: docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+description: "Learn how to contribute additional validation (lint) rules to the Desktop Modeler through lint rule plug-ins."
+---
+
+The Desktop Modeler uses [bpmnlint](https://github.com/bpmn-io/bpmnlint) to validate BPMN diagrams. Through defined modeler extension points you add plug-in additional lint rules or configure/disable existing rules.
```
Comment (christian-konrad) [inline docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:7]: Typo "you add plug-in additional..."

Comment (christian-konrad) [reply docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:7]: Also, better to highlight the value of this feature first (allows to add your own validation rules to ensure processes follow blabla...) before diving into the technical details

File: docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+2. Install the dependencies:
+
+```
+npm install
+```
+
+3. Add a custom rule
+
+To add a custom rule, add it to the [`bpmnlint-plugin-custom/rules`](https://github.com/camunda/camunda-modeler-custom-linter-rules-plugin/tree/master/bpmnlint-plugin-custom/rules) folder. The example project contains a [`no-manual-task.js`](https://github.com/camunda/camunda-modeler-custom-linter-rules-plugin/blob/master/bpmnlint-plugin-custom/rules/no-manual-task.js) file implements a custom rule.
```
Comment (christian-konrad) [inline docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:31]: Typo: "... file THAT implements a custom rule"

**Comment by philippfromme** [comment]
> Let me know if you would like a technical review here 👍

Will let you know after addressing @christian-konrad's comments.

File: docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md (hunk 7 above)
Comment (christinaausley) [reply docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:31]: Noticed this as well -- adjusted in my latest commit.

**Review by christinaausley — DISMISSED** [review]
Adjusted some typos, spacing, and the numerical list in my latest commit. Let me know if you have any questions about these changes.

File: docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md (hunk 6 above)
Comment (christinaausley) [reply docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md]: Reworded this in my latest commit -- is this what you mean?

Comment (christian-konrad) [reply docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md]: @philippfromme @christinaausley more like explaining what those rules are for—why should a user care about it? Maybe also adding a screenshot of how a custom rule will appear in the problems panel, making it easier to correlate for the reader

Comment (philippfromme) [reply docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md]: I was also thinking about adding a screenshot to sell it better. Will make adjustments. I'll also try and discuss this again with @nikku.

**Comment by philippfromme** [comment]
@christinaausley Just to clarify, this PR is not ready yet. I want to address @christian-konrad's comments first. 😃

**Comment by christinaausley** [comment]
@philippfromme Apologies for the premature adjustments -- I saw you had mentioned it's ready to be reviewed and merged and I pushed my review at just about the same time you made the comment about @christian-konrad's comments first. Looks like we may need to merge main in here, but let me know when you'd like me to take a look 👍

**Comment by philippfromme** [comment]
@christinaausley no worries. I've made a few adjustments:

* rephrased the introduction to introduce as little new terminology as possible
* renamed the page to _Custom lint rules_
* added a screenshot (will be automated as part of https://github.com/camunda/camunda-docs-modeler-screenshots/ later)

It's now ready for your final review. 🚀

**Review by nikku — APPROVED** [review]
Looks great + reads great to me.

**Comment by akeller** [comment]
> Immediately, as this feature has already been released.

Quick question here - was this released prior to the docs being written on purpose or by accident?

**Comment by nikku** [comment]
> was this released prior to the docs being written on purpose or by accident?

On purpose. We've already documented the actual thing [on GitHub](https://github.com/camunda/camunda-modeler-custom-linter-rules-plugin) and simply follow up inside the camunda.io docs now.

**Review by christinaausley — DISMISSED** [review]

**Comment by philippfromme** [comment]
@christinaausley Are we allowed to merge?

**Comment by akeller** [comment]
@philippfromme yes please merge when ready! We can even do a release today so the changes are live.

**Comment by philippfromme** [comment]
Sorry @christinaausley and @akeller, can I get another approval? I rebased the branch on master and squashed the commits.

**Review by akeller — APPROVED** [review]

**Comment by philippfromme** [comment]
🚀🚀🚀

**Pull request merged**